packages/components/tests/unit/helpers/validateCssAndMatchSnapshot.ts

```diff
@@ -2,16 +2,24 @@ import { expect } from '@jest/globals';
 
 /*
   TODO: Existing tests for:
-    - !important on host display style
-    - !important on slotted styles
     - @hover media query for hover styles
-    - Host hidden styles
     should also be covered for all getComponentCss conditions
 */
-export const validateCssAndMatchSnapshot = (css: string) => {
-  // We shouldn't use visibility: visible since it cannot be overwritten, use inherit instead
+export const validateCssAndMatchSnapshot = (css: string, isInternal = false) => {
+  // We shouldn't use visibility: visible since it cannot be overridden, use inherit instead
   expect(css).not.toMatch(/visibility:\s*(?:var\(.+,\s*visible\s*\)|visible);/);
   // Invalid css which was produced before
   expect(css).not.toMatch('. {');
+  // Expect no !important rule on display style of :host selector since it should be overridable (Match all display with following !important rule)
+  expect(css).not.toMatch(/:host\s*{\s*[^}]*display:\s*(?:var\(\s*--[\w-]+\s*,\s*[\w-]+\s*\)|[\w-]+)\s*!important/);
```

Comment:
Since `\w` is short for `[A-Za-z0-9_]` I guess this matches characters that will never occur, therefore using `[a-z-]` or `[A-Za-z-]` if capital characters are relevant is more precise.  
Similar story for `\s` which matches tabs and line breaks while the only thing that matters are spaces?
`*` could also be replaced with `?` for the most part when there is a single optional space. Or no quantifier at all, when it is always there, e.g. within `display: xyz`.